plugins/output-kafka/src/main/scala/com/stratio/sparkta/plugin/output/kafka/producer/KafkaProducer.scala

```diff
@@ -0,0 +1,106 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.stratio.sparkta.plugin.output.kafka.producer
+
+import java.io.{Serializable => JSerializable}
+import java.util.Properties
+import com.stratio.sparkta.sdk.ValidatingPropertyMap._
+
+import kafka.producer.{KeyedMessage, Producer, ProducerConfig}
+
+import scala.collection.mutable
+
+trait KafkaProducer {
+  def send(properties: Map[String, JSerializable], topic: String, message: String): Unit = {
+    val keyedMessage: KeyedMessage[String, String] = new KeyedMessage[String, String](topic, message)
+    KafkaProducer.getProducer(topic, properties).send(keyedMessage)
+  }
+
+}
+
+object KafkaProducer {
+  private val DefaultHostPort: String = "localhost:9092"
+  private val DefaultKafkaSerializer: String = "kafka.serializer.StringEncoder"
+  private val DefaultRequiredAcks: String = "0"
+  private val DefaultProducerType = "1"
+  private val DefaultBatchNumMessages = "200"
+
+  private val producers: mutable.Map[String, Producer[String, String]] = mutable.Map.empty
+
+  private val getString: ((Map[String, JSerializable], String, String) => String) = (properties, key, default) => {
+    properties.get(key) match {
+      case Some(v) => v.toString
+      case None => default
+    }
+  }
+
+  private val getList: ((Map[String, JSerializable], String, String) => String) = (properties, key, default) => {
+    val valueAsObject = properties.getMapFromJsoneyString(key)
+    val valueAsSeq = valueAsObject.map(c =>
+      (c.get("host") match {
+        case Some(value) => value.toString
+        case None => default.split(":")(0)
+      },
+        c.get("port") match {
+          case Some(value) => value.toString.toInt
+          case None => default.split(":")(1)
+        }))
+
+    (for (elem <- valueAsSeq) yield s"${elem._1}:${elem._2}").mkString(",")
+  }
+
+  private val mandatoryOptions: Map[String, ((Map[String, JSerializable], String, String) => AnyRef, String)] = Map(
+    "metadata.broker.list" ->(getList, DefaultHostPort),
+    "serializer.class" ->(getString, DefaultKafkaSerializer),
+    "request.required.acks" ->(getString, DefaultRequiredAcks),
+    "producer.type" ->(getString, DefaultProducerType),
+    "batch.num.messages" ->(getString, DefaultBatchNumMessages))
+
+
+  def extractOptions(properties: Map[String, JSerializable],
+                     map: Map[String, ((Map[String, JSerializable], String, String) => AnyRef, String)]): Properties = {
+    val props = new Properties()
+    for {(key, (func, default)) <- map} {
+      properties.get(key) match {
+        case Some(value) => props.put(key, func(properties, key, default))
+        case None =>props.put(key, default)
+      }
```

Comment:
Java code??? 

You can change for something like this:

map.foreach{ case (key, (func, default)) =>
      properties.get(key).foreach(value => props.put(...)
    }